scripts/draw_container/draw_container.gml

```diff
@@ -449,41 +465,94 @@ function draw_container(data)
+		var left_width = 0;
+		var right_width = line_width;
+		
+		for (var i = 0; i < array_length(lines[line_index].segments); i++) {
 			
-			//TODO wonky fade outs with all of these shadows
+			var segment = lines[line_index].segments[i];
 			
-			//thin perp shadows
-			draw_text_ext(startX - 1, startY, data.str, data.lineHeight, data.renderWidth);
-			draw_text_ext(startX + 1, startY, data.str, data.lineHeight, data.renderWidth);
-			draw_text_ext(startX, startY - 1, data.str, data.lineHeight, data.renderWidth);
-			draw_text_ext(startX, startY + 1, data.str, data.lineHeight, data.renderWidth);
+			var text_to_draw = segment.contents;
+			//text_to_draw = string_replace_all(text_to_draw," ","_");
 			
-			//diag thicker shadows
-			draw_text_ext(startX - 1, startY - 1, data.str, data.lineHeight, data.renderWidth);
-			draw_text_ext(startX - 1, startY + 1, data.str, data.lineHeight, data.renderWidth);
-			draw_text_ext(startX + 1, startY - 1, data.str, data.lineHeight, data.renderWidth);
-			draw_text_ext(startX + 1, startY + 1, data.str, data.lineHeight, data.renderWidth);
-		}
+			if (text_to_draw != "")
+			{
+				
+				var size_of_segment = string_width(text_to_draw);
+				right_width -= string_width(text_to_draw);
+				var width_offset = get_width_offset(left_width, right_width);
+				
+				text_to_draw = string_repeat("\n", line_index) + text_to_draw + string_repeat("\n", (total_lines - line_index));
+				
+				if (data.shadowAlpha != 0 && data.computedOpacity != 0)
+				{
+					draw_set_alpha(data.shadowAlpha * data.computedOpacity);
+					draw_set_color(data.shadowColor);
+			
+					//TODO wonky fade outs with all of these shadows
+			
+					//thin perp shadows
+					draw_text_ext(startX + width_offset - 1, startY, text_to_draw, data.lineHeight, data.renderWidth);
+					draw_text_ext(startX + width_offset + 1, startY, text_to_draw, data.lineHeight, data.renderWidth);
+					draw_text_ext(startX + width_offset, startY - 1, text_to_draw, data.lineHeight, data.renderWidth);
+					draw_text_ext(startX + width_offset, startY + 1, text_to_draw, data.lineHeight, data.renderWidth);
+			
+					//diag thicker shadows
+					draw_text_ext(startX - 1 + width_offset, startY - 1, text_to_draw, data.lineHeight, data.renderWidth);
+					draw_text_ext(startX - 1 + width_offset, startY + 1, text_to_draw, data.lineHeight, data.renderWidth);
+					draw_text_ext(startX + 1 + width_offset, startY - 1, text_to_draw, data.lineHeight, data.renderWidth);
+					draw_text_ext(startX + 1 + width_offset, startY + 1, text_to_draw, data.lineHeight, data.renderWidth);
+				}
 
-		if (data.textAlpha != 0 && data.computedOpacity != 0)
-		{
-			draw_set_alpha(data.textAlpha * data.computedOpacity);
-			draw_set_color(data.textColor);
+				if (data.textAlpha != 0 && data.computedOpacity != 0)
+				{
+					draw_set_alpha(data.textAlpha * data.computedOpacity);
+					if (array_length(segment.tags) == 0) {
+						draw_set_color(data.textColor);
+					} else {
```

Comment:
There are many places of inconsistent formatting.  This project uses whitespace liberally.  For example:
```
if (expression)
{
    someCode
}
else if (anotherExpression)
{
    someMoreCode
}
else
{
    evenMoreCode
}
```
One rule (whether spoken, using a linter, or unspoken) when working on projects is to conform to that projects formatting style.  It's a pain (I have to use a style for my day job that isn't what I like), but it is best for the project and its future readers to be consistent.

Reply:
Found and replaced all instances of a `) {` `else {` or } else` pair to remove Egyptian braces.
If there are other formatting corrections you'd like to see, or anything I missed, let me know.